src/diffusers/schedulers/scheduling_euler_ancestral_discrete.py

```diff
@@ -150,7 +150,11 @@ def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.devic
         sigmas = np.interp(timesteps, np.arange(0, len(sigmas)), sigmas)
         sigmas = np.concatenate([sigmas, [0.0]]).astype(np.float32)
         self.sigmas = torch.from_numpy(sigmas).to(device=device)
-        self.timesteps = torch.from_numpy(timesteps).to(device=device)
+        if str(device).startswith("mps"):
```

Comment:
Can it be `mps:1`? :exploding_head:

Reply:
Exactly, I found `mps:0` when testing on my computer :)

Reply:
In other places we use `device.type == "mps"`, but the method signature allows strings too.